semver-breaking: "Unused variable" errors with deny(unused).

The following used to build fine with defmt 0.2.1, and now fails with defmt 0.2.2. This is due to #477 

```rust
#![deny(unused)]
let x = 1; // error: unused variable: `x` if no "defmt-xxx" features are enabled
info!("x={}", x);
```

My idea would be to just match the `args`. This should look sth like following but I am struggeling with the macro syntax:
```rust
cfg_if::cfg_if! {
    if #[#logging_enabled] {
        ...
    } else {
        match (#(&(#args)),*) {
            _ => {}
        }
    }
}
```
Currently I am getting a `error[E0433]: failed to resolve: use of undeclared crate or module 'cfg_if'`, although I specified it as a dependency for `defmt-macros`.

--- a/macros/src/lib.rs
+++ b/macros/src/lib.rs
@@ -480,11 +480,9 @@ fn cfg_if_logging_enabled(level: Level) -> TokenStream2 {
     let features_release = necessary_features_for_level(level, false);
 
     quote!(
-        cfg(
-            any(
-                all(debug_assertions, any(#( feature = #features_dev ),*)),
-                all(not(debug_assertions), any(#( feature = #features_release ),*))
-            )
+        any(
+            all(    debug_assertions,  any(#( feature = #features_dev     ),*)),
+            all(not(debug_assertions), any(#( feature = #features_release ),*))
         )
     )
 }
@@ -513,7 +511,7 @@ fn log(level: Level, log: FormatArgs) -> TokenStream2 {
     let sym = mksym(&ls, level.as_str(), true);
     let logging_enabled = cfg_if_logging_enabled(level);
     quote!({
-        #[#logging_enabled] {
+        #[cfg(#logging_enabled)] {
             match (#(&(#args)),*) {
                 (#(#pats),*) => {
                     if let Some(mut _fmt_) = defmt::export::acquire() {
@@ -524,6 +522,11 @@ fn log(level: Level, log: FormatArgs) -> TokenStream2 {
                     }
                 }
             }
+        }
+        // if logging is disabled match args, so they are not unused
+        #[cfg(not(#logging_enabled))]
+        match (#(&(#args)),*) {
+            _ => {}
         }
     })
 }

